src/frontend/src/flows/verifiableCredentials/vcIssuer.ts

```diff
@@ -107,4 +109,51 @@ export class VcIssuer {
 
     return result.Ok;
   };
+
+  getDerivationOrigin = async ({
+    origin,
+  }: {
+    origin: string;
+  }): Promise<
+    | { kind: "origin"; origin: string }
+    | { kind: "use_default" }
+    | { kind: "error" }
+  > => {
+    const actor = await this.createActor();
+
+    const methodName = "derivation_origin" as const;
+
+    let result;
+    try {
+      result = await actor[methodName]({
+        frontend_hostname: origin,
+      });
+    } catch (e: unknown) {
+      try {
+        // XXX: this "ensures" that the error is the one expected by the issuer spec.
+        // This unfortunately depends on the exact method name, only works for query calls and
```

Comment:
The standard doesn't say it should be marked a query call: https://github.com/dfinity/internet-identity/blob/36621c439f3eef17ed4c2fba9b2f8160b656b95e/docs/vc-spec.md#L4

Is that a problem? Can any method be called as query even it's not marked as query?

Reply:
good catch, that's a typo. I meant "update".

Reply:
Mh actually I'm not sure what you mean. This is an update call, right? and called as such in this comment?

Reply:
Done!